### PR TITLE
Add support for FCM

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Adds configuration options for Firebase Cloud Messaging
- Refactors Delivery package to support both APNS and FCM delivery (and one or the other)
- Updates server startup params to correctly pass in port
- Simplifies API tests to use standard `Stop()` method
- Adds support for connecting to XMTP nodes via TLS

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -58,14 +58,28 @@ func main() {
...
 
 	if opts.Xmtp.ListenerEnabled {
-		deliveryService, err := delivery.NewDeliveryService(logger, opts.Apns)
-		if err != nil {
-			log.Fatalf("Failed to initialize delivery service")
+		var apns *delivery.ApnsDelivery
+		var fcm *delivery.FcmDelivery
+		var err error
+
+		if opts.Apns.KeyId != "" {
```
Comment (neekolas) [inline cmd/server/main.go:65]: Will add some more advanced configuration validation in a subsequent PR

File: pkg/delivery/fcm.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+func (f *FcmDelivery) Send(ctx context.Context, token, topic, message string) error {
+	title := "New message from XMTP"
+	body := "Open app to read"
+	data := map[string]string{
+		"topic":            topic,
+		"encryptedMessage": message,
+	}
+
+	_, err := f.client.Send(ctx, &messaging.Message{
```
Comment (neekolas) [inline pkg/delivery/fcm.go:50]: I tried just throwing the kitchen sink at this with all possible arguments. Will test with a real FCM account and see if it actually allows me to send this whole payload if the FCM account is only partially setup (for example, configuring APNS options here without having APNS setup in Firebase dashboard). If that doesn't work, will have to create a more involved setup where you can configure what services are included in the payload or not.

File: cmd/server/main.go [hunk 3]
```diff
@@ -58,20 +58,34 @@ func main() {
...
 		if err != nil {
-			log.Fatalf("failed to initialize listener")
+			logger.Fatal("failed to initialize listener", zap.Error(err))
 		}
 		listener.Start()
 	}
 
 	if opts.Api.Enabled {
-		apiServer = api.NewApiServer(logger, installationsService, subscriptionsService)
+		apiServer = api.NewApiServer(logger, opts.Api.Port, installationsService, subscriptionsService)
```
Comment (mkobetic) [inline cmd/server/main.go:88]: Would it make sense to pass in `opts.Api` instead of just the port?

File: pkg/xmtp/client.go [hunk 4]
```diff
@@ -2,24 +2,35 @@ package xmtp
...
+		grpc.WithTransportCredentials(getCredentials(useTls)),
 	)
 }
 
-func NewClient(ctx context.Context, apiAddress string) (v1.MessageApiClient, error) {
-	conn, err := newConn(ctx, apiAddress)
+func getCredentials(useTls bool) credentials.TransportCredentials {
+	if useTls {
+		return credentials.NewTLS(&tls.Config{
+			InsecureSkipVerify: true,
```
Comment (mkobetic) [inline pkg/xmtp/client.go:26]: Why do we need this?

**Review by mkobetic — APPROVED** [review]
Minor comments, but otherwise LGTM

File: pkg/xmtp/client.go (hunk 4 above)
Comment (neekolas) [reply pkg/xmtp/client.go:26]: Good call out. I double checked and we don't actually need it.

File: cmd/server/main.go (hunk 3 above)
Comment (neekolas) [reply cmd/server/main.go:88]: Sure. It does give us some more flexibility to add options later.

**Pull request merged**